### ISSUE TITLE
RDS snapshot restore ignores specified engine_version

### ISSUE THREAD

**Issue by maxleonca** [terraform-providers/terraform-provider-aws#7723]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
0.11.11

aws provider version 1.59.0

### Affected Resource(s)

RDS Postgres instance

### Terraform Configuration Files

```hcl
data "aws_db_snapshot" "db_prod_snapshot" {
    most_recent = true
    db_instance_identifier = "production"
}

resource "aws_db_instance" "staging_db" {
    identifier = "staging-vpc"

    snapshot_identifier = "${data.aws_db_snapshot.db_prod_snapshot.id}"
    skip_final_snapshot = true
    instance_class = "${var.db_instance_class}"
    auto_minor_version_upgrade = true
    engine_version = "9.4.20"
    apply_immediately = true

    db_subnet_group_name = "${var.db_subnet_group}"
    vpc_security_group_ids = "${var.rds_security_groups}"

    backup_retention_period = 0

    tags {
    }
}
  
```

### Debug Output

https://gist.github.com/maxleonca/c242d786da61cb9df643be120725c90b

### Panic Output

none

### Expected Behavior

The new engine version specified in the code should have been applied to the restored snapshot

### Actual Behavior

The specified engine version is ignored and the one that was on the original DB from where the snapshot was taken is enforced during the restoration process.

### Steps to Reproduce

Import a snapshot and change the engine version.

1. `terraform apply`

### References

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.1.0 of the Terraform AWS Provider, likely middle of this week.

**Comment by bflad** [comment]
This has been released in [version 2.1.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#210-march-07-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
